src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -0,0 +1,143 @@
+
+            // Since I don't know how to check anything with parentFleetMember yet, lets just leave it out of the equation for now
+            isExpected && hasNotInstalledOnThisVariantAlready
+        } else {
+            AnonymousLogger.log("shouldInstallHullmodExoticToVariant()\tthere was no install data for this key ${hullmodExoticKey}", "HullmodExoticHandler")
+            // If we don't have any install data in the map, this is going to be simple. The optional variant list must be present
+            // If not present, just... do nothing for now, because we can't start off from a submodule I hope. So for now, throw
+            if (variantList.isPresent().not()) {
+                throw IllegalStateException("Since there is no HullmodExoticInstallData for these parameters, the Optional must be present!")
+            }
+
+            // And now, add stuff to the map ...
+            lookupMap[hullmodExoticKey] = HullmodExoticInstallData(
+                    parentFleetMemberAPI = parentFleetMember,
+                    listOfExpectedVariants =  variantList.get(),
+                    listOfVariantsWeInstalledOn = emptyList()
+            )
+
+            // And obviously return true
+            true
+        }
+
+        AnonymousLogger.log("<-- shouldInstallHullmodExoticToVariant()\treturning ${retVal}", "HullmodExoticHandler")
+        return retVal
+    }
+
+
+    fun installHullmodExoticToVariant(hullmodExotic: HullmodExotic, parentFleetMember: FleetMemberAPI, variant: ShipVariantAPI): Boolean {
+        AnonymousLogger.log("--> installHullmodExoticToVariant()", "HullmodExoticHandler")
+        // Basically, this should be easy:
+        // 1. form up the key
+        // 2. grab the InstallData. Throw if there is none.
+        // 3. grab the already-installed list
+        // 4. throw if we're already in it
+        // 5. convert it to mutable list
+        // 6. add the variant to it
+        // 7. make a copy of the original lookup[key] value with the updated already-installed list
+        // 8. update the map with new installed data
+        val hullmodExoticKey = HullmodExoticKey(
+                hullmodExotic = hullmodExotic,
+                parentFleetMember = parentFleetMember
+        )
+
+        val currentInstallData = lookupMap[hullmodExoticKey]
+        if (currentInstallData == null) {
+            throw IllegalStateException("We should have had this key in the lookup map!!! lookupMap: ${lookupMap}")
+        }
+
+        val alreadyInstalledList = currentInstallData.listOfVariantsWeInstalledOn
+        // just check
+        val alreadyIn = alreadyInstalledList.contains(variant)
+        if (alreadyIn) {
+            // Actually, lets not throw just yet
+            // TODO change to throwing
+            AnonymousLogger.log("installHullmodExoticToVariant()\t\t !!!!! Trying to install on a variant that's already been installed on, returning false!!!!!", "HullmodExoticHandler")
```

Comment:
i'm seeing this happening in the logs, so lets **not** change it to throwing.
changing to throwing can be done *once* we start checking if it should be installed (which we're doing) and skipping those that shouldn't be installed on. which is, what i think ,also already been done.

hmm...

Reply:
we will not change to throwing because of reasons encountered during #38  - resolving.